### PR TITLE
Solution in C++ for Stuck in a Rut (usaco-1061)

### PR DESCRIPTION
**Pull request by CELLSecret1121**
Wrote solution (with comments) in C++ for Stuck in a Rut.

_If any of the below doesn't apply to this Pull Request, mark the checkbox as
done._

- [x] I have tested my code
- [x] I have followed the code style guidelines mentioned
      [here](https://usaco.guide/general/adding-solution/#code-conventions)
- [x] I have properly formatted the files according to the steps mentioned
      [here](https://usaco.guide/general/adding-solution#steps)

### CODE REVIEW

**Comment by SansPapyrus683** [comment]
> I have followed the code style guidelines mentioned [here](https://usaco.guide/general/adding-solution#code-conventions)

are-you-sure-john-cena

**Comment by CELLSecret1121** [comment]
wait which one did I not meet
I'll fix it

**Comment by SansPapyrus683** [comment]
read the thing again

**Comment by CELLSecret1121** [comment]
So I cannot use a vector?

Screen Shot 2021-12-23 at 11 51 50

**Comment by SansPapyrus683** [comment]
> So I cannot use a vector?
> 
> Screen Shot 2021-12-23 at 11 51 50
> 
> (yes the code would be way more complicated than it needs to be without vector)

var length array != vector

**Comment by CELLSecret1121** [comment]
So is it good enough now? What else do I need to edit?

**Comment by SansPapyrus683** [comment]
use tabs to indent

**Comment by CELLSecret1121** [comment]
I believe I did that.

**Comment by CELLSecret1121** [comment]
Done. Can you check thanks

apparently tabs got converted to spaces when I copied the code from another IDE.

File: solutions/usaco-1061.mdx [hunk 1]
```diff
@@ -24,6 +24,89 @@ Now we can sort cows by their static coordinates, which ensures that earlier col
...
+
+	// sort east cows by y value
+	sort(Ecows.begin(), Ecows.end(), [](auto &left, auto &right) {
+		return left[1] < right[1];
+	});
+
+	// we sort them by x/y value because we know that, for a single cow, 
+	// it reaches a lower x/y value and then a higher one. 
+	// Thus, it is more we would not miss any earlier collisions
+	// if we iterate from lower to a higher value.
```
Comment (SansPapyrus683) [inline solutions/usaco-1061.mdx:61]:
```cpp
/*
 * format multiline comments like this
 * don't use a bunch of single line comments
 */
```

File: solutions/usaco-1061.mdx [hunk 2]
```diff
@@ -24,6 +24,89 @@ Now we can sort cows by their static coordinates, which ensures that earlier col
...
 **Time Complexity:** $\mathcal{O}(N^2)$
 <LanguageSection>
+<CPPSection>
+```cpp
+#include <bits/stdc++.h>
+using namespace std;
+
+int main() {
+	int N; cin >> N;
+	vector<array<int, 3>> Ncows, Ecows;
```
Comment (SansPapyrus683) [inline solutions/usaco-1061.mdx:34]: rename to `n_cows` and `e_cows`

File: solutions/usaco-1061.mdx [hunk 3]
```diff
@@ -24,6 +24,89 @@ Now we can sort cows by their static coordinates, which ensures that earlier col
...
+
+	// stores the amount of grass it eats
+	int stop[N];
+	for (int i = 0; i < N; i++) {
+		stop[i] = -1;
+  	}
+
+
+	for (auto Ncow : Ncows) {
+    		for (auto Ecow : Ecows) {
```
Comment (SansPapyrus683) [inline solutions/usaco-1061.mdx:71]: `n_cos` and `e_cow`
also i'd like if you didn't use auto

**Review by SansPapyrus683 — CHANGES_REQUESTED** [review]

File: solutions/usaco-1061.mdx [hunk 4]
```diff
@@ -24,6 +24,89 @@ Now we can sort cows by their static coordinates, which ensures that earlier col
...
+		return left[1] < right[1];
+	});
+
+	// we sort them by x/y value because we know that, for a single cow, 
+	// it reaches a lower x/y value and then a higher one. 
+	// Thus, it is more we would not miss any earlier collisions
+	// if we iterate from lower to a higher value. 
+
+	// stores the amount of grass it eats
+	int stop[N];
```
Comment (SansPapyrus683) [inline solutions/usaco-1061.mdx:63]: use vector

Comment (CELLSecret1121) [reply solutions/usaco-1061.mdx:63]: wait so `vector<int> stop`? It's not really necessary but I'll do it

**Comment by CELLSecret1121** [comment]
fixed (I think)

File: solutions/usaco-1061.mdx [hunk 5]
```diff
@@ -24,6 +24,89 @@ Now we can sort cows by their static coordinates, which ensures that earlier col
...
+			} 
+		}
+	}
+
+	/* output the final value; if the value is still -1, 
+  	 * or the default value, it goes infinitely.
+	 */
+  	for (int i = 0; i < N; i++) {
+    		if (stop[i] == -1) {
+      			cout << "Infinity\n";
```
Comment (ryanchou-dev) [inline solutions/usaco-1061.mdx:101]:
```suggestion
      			cout << "Infinity" << '\n';
```
consistency

File: solutions/usaco-1061.mdx [hunk 6]
```diff
@@ -24,6 +24,89 @@ Now we can sort cows by their static coordinates, which ensures that earlier col
...
+    		} else {
+      			cout << stop[i] << '\n';
+    		}
+  	}
+} // by CELLSecret
```
Comment (ryanchou-dev) [inline solutions/usaco-1061.mdx:106]: remove this

**Review by ryanchou-dev — CHANGES_REQUESTED** [review]
rest is ok

File: solutions/usaco-1061.mdx (hunk 6 above)
Comment (ryanchou-dev) [reply solutions/usaco-1061.mdx:106]: you can list yourself as an author instead

**Comment by CELLSecret1121** [comment]
So what do I need to do now?

File: solutions/usaco-1061.mdx [hunk 7]
```diff
@@ -2,7 +2,7 @@
 id: usaco-1061
 source: USACO Bronze 2020 December
 title: Stuck in a Rut
-author: Ryan Chou, Alex Yang
+author: Ryan Chou, Alex Yang, CELLSecret
```
Comment (SansPapyrus683) [inline solutions/usaco-1061.mdx:5]: please use real name

File: solutions/usaco-1061.mdx [hunk 8]
```diff
@@ -24,6 +24,90 @@ Now we can sort cows by their static coordinates, which ensures that earlier col
...
+
+	// stores the amount of grass it eats
+	vector<int> stop(N);
+	for (int i = 0; i < N; i++) {
+		stop[i] = -1;
+  	}
+
+
+	for (auto n_cow : n_cows) {
+    		for (auto e_cow : e_cows) {
```
Comment (SansPapyrus683) [inline solutions/usaco-1061.mdx:73]: can you not use auto

**Review by SansPapyrus683 — CHANGES_REQUESTED** [review]

File: solutions/usaco-1061.mdx (hunk 8 above)
Comment (CELLSecret1121) [reply solutions/usaco-1061.mdx:73]: I don't think I can change it... `array` doesn't work for some reason

Comment (SansPapyrus683) [reply solutions/usaco-1061.mdx:73]: k then

**Review by SansPapyrus683 — APPROVED** [review]

File: solutions/usaco-1061.mdx (hunk 3 above)
Comment (Aadit-Ambadkar) [reply solutions/usaco-1061.mdx]: auto is fine

**Review by Aadit-Ambadkar — APPROVED** [review]
I'll approve but this is more so a contributor thing than an author thing (since its just uploading code)

**Comment by CELLSecret1121** [comment]
Yea I thought of that too but I don't see a contributor row and ryanchou-dev and SansPapyrus683 told me to put it as author. You can change it if you want.

**Comment by ryanchou-dev** [comment]
> I'll approve but this is more so a contributor thing than an author thing (since its just uploading code)

it's a solution. there are no contributors.

**Review by ryanchou-dev — APPROVED** [review]

**Pull request merged**